src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunReader.cs

```diff
@@ -557,6 +567,23 @@ private unsafe void EnsureManifestReferences()
             }
         }
 
+        private void EnsureManifestReferenceNames()
+        {
+            if (_manifestReferenceNames != null)
+            {
+                return;
+            }
+            EnsureManifestReferences();
+            _manifestReferenceNames = new List<string>(_manifestReferences.Count);
```

Comment:
Do we need to keep the names and indices separately? Can't we use just the dictionary and get the names from it using the Dictionary.Keys property?

Reply:
I think we can do that. I was originally reluctant to pursue this due to introducing the inefficiency of having to sort the list by indices when we display the manifest metadata section (as enumeration of a dictionary isn't guaranteed to follow the ordering of insertions) but I don't think it's happening on any perf-critical path; I'll double-check that.